webapp/tests/karma/ts/modules/contacts/contacts.component.spec.ts

```diff
@@ -0,0 +1,395 @@
+      expect(searchService.search.callCount).to.equal(2);
+      expect(searchService.search.args[1]).to.deep.equal(
+        [
+          'contacts',
+          { types: { selected: ['childType'] } },
+          { limit: 50 },
+          {
+            displayLastVisitedDate: true,
+            visitCountSettings: { monthStartDate: false, visitCountGoal: 1 },
+          },
+          undefined,
+        ]
+      );
+    }));
+
+    it('always saves default sort', fakeAsync(() => {
+      authService.has.resolves(true);
+      settingsService.get.resolves({
+        uhc: {
+          contacts_default_sort: 'something',
+          visit_count: {
+            month_start_date: false,
+            visit_count_goal: 1,
+          },
+        },
+      });
+      component.ngOnInit();
+      flush();
+
+      expect(authService.has.callCount).equal(2);
+      expect(authService.has.args[1]).to.deep.equal(['can_view_last_visited_date']);
+      expect(component.lastVisitedDateExtras).to.equal(true);
+      expect(component.visitCountSettings).to.deep.equal({
+        monthStartDate: false,
+        visitCountGoal: 1,
+      });
+      expect(component.sortDirection).to.equal('something');
+      expect(searchService.search.callCount).to.equal(2);
+      expect(searchService.search.args[1]).to.deep.equal(
+        [
+          'contacts',
+          { types: { selected: ['childType'] } },
+          { limit: 50 },
+          {
+            displayLastVisitedDate: true,
+            visitCountSettings: { monthStartDate: false, visitCountGoal: 1 },
+          },
+          undefined,
+        ]
+      );
+    }));
+
+    it('saves uhc default sorting', fakeAsync(() => {
+      authService.has.resolves(true);
+      settingsService.get.resolves({
+        uhc: {
+          contacts_default_sort: 'last_visited_date',
+          visit_count: {
+            month_start_date: 25,
+            visit_count_goal: 125,
+          },
+        },
+      });
+      component.ngOnInit();
+      flush();
+
+      expect(authService.has.callCount).equal(2);
+      expect(authService.has.args[1]).to.deep.equal(['can_view_last_visited_date']);
+      expect(component.lastVisitedDateExtras).to.equal(true);
+      expect(component.visitCountSettings).to.deep.equal({
+        monthStartDate: 25,
+        visitCountGoal: 125,
+      });
+      expect(component.sortDirection).to.equal('last_visited_date');
+      expect(component.defaultSortDirection).to.equal('last_visited_date');
+      expect(searchService.search.callCount).to.equal(2);
+      expect(searchService.search.args[1]).to.deep.equal(
+        [
+          'contacts',
+          { types: { selected: ['childType'] } },
+          { limit: 50 },
+          {
+            displayLastVisitedDate: true,
+            visitCountSettings: { monthStartDate: 25, visitCountGoal: 125 },
+            sortByLastVisitedDate: true,
+          },
+          undefined,
+        ]
+      );
+    }));
+  });
```

Comment:
There are lots of tests that are missing from this file, that should be migrated from the original file.

Reply:
Yes. There are some tests I have left out and plan on migrating them with the next PR that migrate the content area.